src/postamble.js

```diff
@@ -119,6 +119,7 @@ function callMain(args) {
 #if ASSERTIONS
   assert(runDependencies == 0, 'cannot call main when async dependencies remain! (listen on Module["onRuntimeInitialized"])');
   assert(__ATPRERUN__.length == 0, 'cannot call main when preRun functions remain to be called');
+  assert(!calledMain, 'in the C world, main() is special in that it should only be called once');
```

Comment:
Maybe mention `callMain` itself as the thing that can't be repeatedly called?

Ironically, although its not advised, calling `_main` directly does't have any of the same issues.

Should we link to an FAQ entry that can help folks who want the semantics of a multi-call main?  What should they do instead?

Reply:
Perhaps an FAQ entry is better than this, yeah. Could mention MODULARIZE etc.

Or, perhaps we could try to fix callMain to be runnable multiple times.

Reply:
Yes I guess there are two solutions we would want to recommend:

1. Use `MODULARIZE` and instantiate a new module for each call
2. Convert to a library that exposes the same functionality as `main` under a different name which can be multi-called.

Reply:
There is also a problem with hard-to-obtain exit code